File: packages/token-app-api/src/database/models/Bankcard.js
const baseSchema = require("../schemas/baseSchema");

module.exports = (sequelize, DataTypes) => {
  const Bankcard = sequelize.define("Bankcard",
    {
      ...baseSchema,
      name: {
        filed: 'name',
        type: DataTypes.STRING,
        allowNull: false,
        length: 20,
      },
      cardNumber: {
        field: 'card_number',
        type: DataTypes.STRING,
        allowNull: false,
        comment: '銀行卡號',
      },
      status: {
        field: 'status',
        type: DataTypes.ENUM([0, 1, 2]),
        defaultValue: 1,
        comment: '銀行卡狀態: 0: 凍結, 1: 啟用, 2: 刪除',
        get() {
          const rawValue = this.getDataValue('status');
          return parseInt(rawValue);
        }
      },
      branchName: {
        filed: 'branch_name',
        type: DataTypes.STRING,
        allowNull: false,
        length: 20,
      },
    }, {
    sequelize,
    paranoid: true, //允許軟刪除
    tableName: "bank_cards",
    timestamps: false,
    underscored: true,
    freezeTableName: true,
  });

  Bankcard.associate = function (models) {
    // associations can be defined here
    Bankcard.belongsTo(models.User, {
      as: "user",
      foreignKey: 'user_id',
      onUpdate: 'CASCADE',
      onDelete: 'CASCADE',
    });

    Bankcard.belongsTo(models.Bank, {
      as: "bank",
      foreignKey: 'bank_id',
      onUpdate: 'CASCADE',
      onDelete: 'CASCADE',
    });

    Bankcard.hasMany(models.PendingOrder, {
      as: 'bankcards',
      foreignKey: {
        name: "bank_card_id",
      },
      onUpdate: 'CASCADE',
      onDelete: 'CASCADE',
    });
    // User.hasMany(models.Comment, {
    //   foreignKey: 'userId',
    //   as: 'comments',
    //   onDelete: 'CASCADE',
    // });
  };
  return Bankcard;
};

File: pkg/models/bank_card.go
package models

import (
	"time"
)

/**
 * @brief BankCard 銀行卡資料模型
 * @description 對應資料庫 bank_cards 表
 */
type BankCard struct {
	ID         int        `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	UserID     *int       `gorm:"column:user_id" json:"user_id,omitempty"`
	BankID     *int       `gorm:"column:bank_id" json:"bank_id,omitempty"`
	Name       string     `gorm:"column:name;type:varchar(255);not null" json:"name"`               // 銀行卡擁有者姓名
	CardNumber string     `gorm:"column:card_number;type:varchar(255);not null" json:"card_number"` // 銀行卡號
	Status     string     `gorm:"column:status;type:varchar(255);not null" json:"status"`           // 0: 正常, 1: 停用, 2: 凍結
	BranchName string     `gorm:"column:branch_name;type:varchar(255);not null" json:"branch_name"` // 分行名稱
	CreatedAt  time.Time  `gorm:"column:created_at;type:timestamptz;not null" json:"created_at"`
	UpdatedAt  time.Time  `gorm:"column:updated_at;type:timestamptz;not null" json:"updated_at"`
	DeletedAt  *time.Time `gorm:"column:deleted_at;type:timestamptz" json:"deleted_at,omitempty"`

	// 關聯
	User *User `gorm:"foreignKey:UserID;references:ID" json:"user,omitempty"`
	Bank *Bank `gorm:"foreignKey:BankID;references:ID" json:"bank,omitempty"`
}

/**
 * @brief TableName 指定資料表名稱
 * @return string 資料表名稱
 */
func (BankCard) TableName() string {
	return "bank_cards"
}
